### PR TITLE
MTM-44862: Use Spring Framework version 5.3.18

### PR DESCRIPTION
**Pull request by kaczynskid-sag**
Use Spring Framework version 5.3.18 and Spring Boot 2.5.12, which address the RCE vulnerability.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -56,7 +56,7 @@
         <microemu.version>2.0.4</microemu.version>
         <mockito.version>3.12.4</mockito.version>
         <slf4j.version>1.7.32</slf4j.version>
-        <spring.version>5.3.16</spring.version>
+        <spring-framework.version>5.3.18</spring-framework.version>
```
Comment (kaczynskid-sag) [inline pom.xml:60]: Aligns the name of the property with the one in `spring-boot-dependencies`.

**Review by lukaszsoszynskisag — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]
Final question: Why not directly jump to 5.2.20?

**Comment by kaczynskid-sag** [comment]
@sag-eweingaertner because `5.3.18` > `5.2.20` ;)

**Review by jkot-c8y — DISMISSED** [review]

**Comment by c8y-kh4mell** [comment]
Hope it was already tested by building and running at least most important microservices

File: java-client/pom.xml [hunk 2]
```diff
@@ -127,13 +127,13 @@
         <dependency>
             <groupId>org.springframework</groupId>
             <artifactId>spring-context</artifactId>
-            <version>${spring.version}</version>
```
Comment (c8y-kh4mell) [inline java-client/pom.xml:130]: Please double check if this rename could somehow downgrade spring dependency for some of agents. We had such situation with CEP Esper in the past.

```
$ grep . -rnw -e "spring.version"
./lwm2m-agent/lwm2m-decoders/api/pom.xml:43:            <version>${spring.version}</version>
./lwm2m-agent/lwm2m-decoders/api/pom.xml:48:            <version>${spring.version}</version>
./lwm2m-agent/lwm2m-integration-tests/pom.xml:25:        <spring.version>4.3.20.RELEASE</spring.version>
```

**Comment by ghost** [comment]
> @sag-eweingaertner because `5.3.18` > `5.2.20` ;)

Sorry, my bad :)

File: java-client/pom.xml (hunk 2 above)
Comment (kaczynskid-sag) [reply java-client/pom.xml:130]: Added backward compatibility property rewrite to allow for independent change in agents and microservices.

**Review by ghost — APPROVED** [review]

**Review by c8y-kh4mell — APPROVED** [review]

**Pull request merged**